### PR TITLE
Fixed misconfigured kubelet on containervm image for node-e2e.

### LINKED ISSUE #30412: Container configuration not correct for node-e2e-test
In node-e2e-test on Containervm image, the containers of the pods created by Kubernetes are subcontainers of docker-daemon/docker/.

It means that the runtime resource usage (docker-daemon) is not correct, because all pods are included.

This problem maybe due to container misconfiguration in node-e2e-test.

### PR DESCRIPTION
**Pull request by dchen1107**
Fixed #30412 to unblock @coufon's performance benchmark work.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/30415)

### CODE REVIEW

**Comment by Random-Liu** [comment]
LGTM

File: test/e2e_node/e2e_service.go [hunk 1]
```diff
@@ -252,6 +252,8 @@ func (es *e2eService) startKubeletServer() (*killCmd, error) {
 		cmdArgs = append(cmdArgs,
 			"--runtime-cgroups=/docker-daemon",
 			"--kubelet-cgroups=/kubelet",
+			"--cgroup-root=/",
```
Comment (Random-Liu) [inline test/e2e_node/e2e_service.go]: It seems that this should also be set for systemd.

**Comment by dchen1107** [comment]
The node e2e failed on the new coreos image. But the change here is only touched cgroup of nodes, and I double-checked the cgroup layout on coreos node, it looks correct to me. Also the failed tests are config map related tests, not cgroup related at all. I will loop into it deeper.

File: test/e2e_node/e2e_service.go (hunk 1 above)
Comment (dchen1107) [inline test/e2e_node/e2e_service.go]: on systemd node cgroup_root should set to system.slice, not "/". The change move the initial the block of code "--cgroup-root="/"" just simply breaks container vm image, but not fixing systemd nodes at all.

**Comment by dchen1107** [comment]
unit/integration failure is caused by https://github.com/kubernetes/kubernetes/issues/30377, and should be fixed by https://github.com/kubernetes/kubernetes/pull/30478

**Comment by Random-Liu** [comment]
@k8s-bot test this issue https://github.com/kubernetes/kubernetes/issues/30377.

**Comment by Random-Liu** [comment]
LGTM

**Pull request merged**